File: src/main/java/org/openmedstack/commands/CommandResponse.java
package org.openmedstack.commands;

import org.openmedstack.ICorrelate;

public class CommandResponse implements ICorrelate {
    private final String _targetAggregate;
    private final Integer _version;
    private final String _faultMessage;
    private final String _correlationId;

    public CommandResponse(String targetAggregate, Integer version, String faultMessage, String correlationId) {
        _targetAggregate = targetAggregate;
        _version = version;
        _faultMessage = faultMessage;
        _correlationId = correlationId;
    }

    public static CommandResponse success(DomainCommand command){
        return new CommandResponse(command.getAggregateId(), command.getVersion(), null, null);
    }

    public String getTargetAggregate() {
        return _targetAggregate;
    }

    public Integer getVersion() {
        return _version;
    }

    public String getFaultMessage() {
        return _faultMessage;
    }

    @Override
    public String getCorrelationId() {
        return _correlationId;
    }
}

File: src/main/java/org/openmedstack/commands/CommandResponse.kt
package org.openmedstack.commands

import org.openmedstack.ICorrelate

class CommandResponse(val targetAggregate: String, val version: Int, val faultMessage: String?, override val correlationId: String?) : ICorrelate {

    companion object {
        @kotlin.jvm.JvmStatic
        fun success(command: DomainCommand): CommandResponse {
            return CommandResponse(command.aggregateId, command.version, null, command.correlationId)
        }

        @kotlin.jvm.JvmStatic
        fun error(command: DomainCommand, error: String): CommandResponse {
            return CommandResponse(command.aggregateId, command.version, error, command.correlationId)
        }
    }
}
